source/PluginDev/Assets/GooglePlayGames/Platforms/Native/NativeClient.cs

```diff
@@ -288,10 +288,26 @@ private static void LaunchBridgeIntent(IntPtr bridgedIntent)
         }
         
         private AndroidJavaObject GetApiClient(GameServices services) {
-            return JavaUtils.JavaObjectFromPointer(C.InternalHooks_GetApiClient(services.AsHandle()));
+            //return JavaUtils.JavaObjectFromPointer(GooglePlayGames.Native.Cwrapper.InternalHooks.InternalHooks_GetApiClient(services.AsHandle()));
+            using (var currentActivity = GetActivity()) {
+                using (AndroidJavaClass jc_plus = new AndroidJavaClass("com.google.android.gms.plus.Plus")) {
+                    using (AndroidJavaObject jc_builder = new AndroidJavaObject("com.google.android.gms.common.api.GoogleApiClient$Builder",currentActivity)) {
+                        jc_builder.Call<AndroidJavaObject> ("addApi", jc_plus.GetStatic<AndroidJavaObject>("API"));
+                        jc_builder.Call<AndroidJavaObject> ("addScope", jc_plus.GetStatic<AndroidJavaObject>("SCOPE_PLUS_LOGIN"));
+                        AndroidJavaObject client = jc_builder.Call<AndroidJavaObject> ("build");
+                        client.Call ("connect");
+                        while(!client.Call<bool>("isConnected"))
```

Comment:
In general I do agree with you, but I was going off you kinda approving it in the thread (I didn't have a whole lot to do with this section of the pull request tbh).

The problem here is (and again, I didn't write this code) is that the ApiClient is built long before this, without the option to add in these apis and scopes, unless I'm wrong?

There's no options in `PlayGamesClientConfiguration` nor `PlayGamesClientConfiguration.Builder` as far as I can see, so if you check for authenticated, you almost certainly will be, but it won't be a correct *version* of the `ApiClient`, it'll be one without the `API` and `SCOPE_PLUS_LOGIN`.

If this potentially came with a warning / exception, in fact all the Get... functions, which logged a debug message and returned without doing anything if you're not logged in, would that do?